autoload/sj/yaml.vim

```diff
@@ -139,5 +139,87 @@ function! s:GetChildren(line_no)
   endwhile
   let next_line_no = next_line_no - 1
 
+  " Preserve trailing empty lines
+  while sj#BlankString(getline(next_line_no)) && next_line_no > line_no
+    let next_line_no = next_line_no - 1
+  endwhile
+
   return [sj#GetLines(line_no + 1, next_line_no), next_line_no]
 endfunction
+
+function! s:splitArrayLines(array)
+  let lines = []
+
+  let line = ''
```

Comment:
Would a better name for this variable be `partial_line`? If I understand this correctly, this part is built-up over several chunks, so I feel like it would make it clearer.

Reply:
If you prefer `partial_line` I'll use that!